protos/flyteidl/core/tasks.proto

```diff
@@ -154,4 +154,44 @@ message Container {
     // Ports to open in the container. This feature is not supported by all execution engines. (e.g. supported on K8s but
     // not supported on AWS Batch)
     repeated ContainerPort ports = 7;
+
+    // BETA: This enables use of CoPilot. This makes it possible to to run a completely portable container, that uses inputs and outputs
+    // only from the local file-system and without having any reference to flyteidl. This is supported only on K8s at the moment.
+    // If CoPilot is enabled, then data will be mounted in accompanying directories specified in the CoPilot settings. If the directories 
+    // are not specified, inputs will be mounted onto and outputs will be uploaded from a pre-determined file-system path. Refer to the documentation
+    // to understand the default paths.
+    bool use_copilot = 8;
+
+    // Optional configuration for CoPilot. If not specified, then default values are used.
+    CoPilot copilot_config = 9;
+}
+
+// This configuration allows executing raw containers in Flyte using the Flyte CoPilot system.
+// Flyte CoPilot, eliminates the needs of flytekit or sdk inside the container. Any inputs required by the users container are side-loaded in the input_path
+// Any outputs generated by the user container - within output_path are automatically uploaded.
+message CoPilot {
+  // MetadataFormat decides the encoding format in which the input metadata should be made available to the containers. 
+  // If the user has access to the protocol buffer definitions, it is recommended to use the PROTO format.
+  // JSON and YAML do not need any protobuf definitions to read it
+  // All remote references in core.LiteralMap are replaced with local filesystem references (the data is downloaded to local filesystem)
+  enum MetadataFormat {
+    // JSON / YAML are serialized represnetation of a map[string]primitive. primitive -> int, string, bool, double/float, bytes
+    JSON = 0;
+    YAML = 1;
+    // Proto is a serialized binary of `core.LiteralMap` defined in flyteidl/core
+    PROTO = 2;
+  }
+  // File system path (start at root). This folder will contain all the inputs exploded to a separate file. 
+  // Example, if the input interface needs (x: int, y: blob, z: multipart_blob) and the input path is "/var/flyte/inputs", then the file system will look like
+  // /var/flyte/inputs/inputs.<metadata format dependent -> .pb .json .yaml> -> Format as defined previously. The Blob and Multipart blob will reference local filesystem instead of remote locations 
+  // /var/flyte/inputs/x -> X is a file that contains the value of x (integer) in string format
+  // /var/flyte/inputs/y -> Y is a file in Binary format
+  // /var/flyte/inputs/z/... -> Note Z itself is a directory
+  // More information about the protocol - refer to docs #TODO reference docs here
+  string input_path = 1;
```

Comment:
should it say `input_dir` ?

Reply:
as discussed, k8s convention seems to be path https://kubernetes.io/docs/concepts/storage/volumes/#aws-ebs-example-configuration